### PR TITLE
feat: Configure app through environment variables

### PR DESCRIPTION
**Pull request by marlass**
## Current problems

Until this moment we utilized build in Angular environment approach with file replacement. However recently, when we unified app we use for testing we noticed that it is not a scalable solution in the long term. Each property that we want to make configurable increase amount of configuration 2 times!!
We added multiple backends -> that's 5 options
Let's add option to enable/disable cds -> now we have 10 options
Now let's add b2b/b2c switch -> now 20 options!!!

And we have already PR for Gigya, incoming CPQ.
You can do math.

To make it even worse, for each configuration we often add dedicated configuration for cypress, couple of scripts into `package.json`. That is growing too fast and one knows where each command is used. New comers seeing that `package.json` are intimidated. Not only them, but we also are confused with it.

## Proposal

Let's use standard mechanism for configuration per environment: env variables. One of the principle for twelve factor app: https://12factor.net/config

It's not a complete transition from file replacement to env variables.
We would go back to the the configuration we had at the beggining: `environment.ts` and `environment.prod.ts`. In first one we don't use env variables and for production we get the values from environment.

### Why this approach?

- no need during dev mode to restart start process. Changes in `environment.ts` will be recompiled
- the amount of configuration was mainly added for testing/infrastructure purposes. You rarely need to build different environments locally, and to fulfill this requirement you can easily update values in environment.prod.ts, while you would normally work only with `environment.ts` during development.
- on each CI job/QA server you now need to set the environment variables for it's purpose
- each job/server runs exactly the same commands, only configuration varies between them
- adding new option require 1 new environment variable and desired behavior for that variable in `app.module.ts`

### Plan of attack

1. Add environment variables mechanism to angular configuration, provide default `environment.ts` files, update `package.json` to not change commands behavior (this PR targets this step)
1. Gradually move CI jobs, QA servers to make use of environment variables.
1. Move non essential scripts (release stuff, e2e) from main package.json to designed place (release stuff into one place, e2e only in cypress)
1. Remove common tasks with specific configurations from package.json. Leave only generic build/start/ssr/test options and handle the configuration through env variables.
(target: no more than 15 commands)

### CODE REVIEW

File: ci-scripts/build-for-deploy.sh
Comment (znikola) [inline ci-scripts/build-for-deploy.sh]: Is the URL correct? Shouldn't it be `https://api.c39j2-walkersde1-d4-public.model-t.cc.commerce.ondemand.com`?

File: projects/storefrontapp/src/app/app.module.ts
Comment (znikola) [inline projects/storefrontapp/src/app/app.module.ts:31]: I kind of liked having these inline in the `imports` array :)

File: projects/storefrontapp/src/app/app.module.ts
Comment (znikola) [inline projects/storefrontapp/src/app/app.module.ts:37]: We should merge this branch into `epic/b2b`. Some files to consider:
- https://github.com/SAP/spartacus/pull/7741/files#diff-827d378f5be063c804b5d194b664a979
- https://github.com/SAP/spartacus/pull/7741/files#diff-67849fba1a80b4db7314a20c4f560c38

File: projects/storefrontapp/src/environments/environment.ts
Comment (znikola) [inline projects/storefrontapp/src/environments/environment.ts]: What is the reason for having to explicitly define the `typings.d.ts`?

**Review by znikola — CHANGES_REQUESTED** [review]

File: projects/storefrontapp/src/app/app.module.ts
Comment (marlass) [reply projects/storefrontapp/src/app/app.module.ts:31]: I couldn't get rid of the `position 4` errors :(
I will try again to improve that.

File: projects/storefrontapp/src/app/app.module.ts
Comment (marlass) [reply projects/storefrontapp/src/app/app.module.ts:37]: I will leave that to them to sync with develop after we merge it.

File: projects/storefrontapp/src/environments/environment.ts
Comment (marlass) [reply projects/storefrontapp/src/environments/environment.ts]: Moved that reference to tsconfig

File: package.json
Comment (znikola) [inline package.json]: Do we want to have CDS enabled for b2b?

**Review by znikola — CHANGES_REQUESTED** [review]

File: package.json
Comment (marlass) [reply package.json]: My bad, fixed.

**Review by znikola — DISMISSED** [review]

**Review by znikola — APPROVED** [review]

**Pull request merged**
